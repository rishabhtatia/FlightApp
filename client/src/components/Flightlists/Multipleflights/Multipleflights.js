import React, { useState } from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import clsx from "clsx";
import styles from "./Multipleflights.module.css";
import Flightlist from "../Flightlist/Flighlist";

const MultipleFlights = ({
  flightListing,
  addToFavouriteList,
  addRemoveItem,
  id
}) => {
  const [expandFlag, setExpandFlag] = useState(false);
  const [isDisabled, setDisabled] = useState(false);
  const price = flightListing.reduce((sum, prev) => prev.price + sum, 0);

  return (
    <div className={styles.multipleFlights}>
      <div className="card">
        <div className="card-body">
          <div className="row">
            <div className="col-sm">
              <h4>Multiple</h4>
              <button
                className={clsx(styles.expandFlagButton, styles.cursorPointer)}
                onClick={() => setExpandFlag(!expandFlag)}
              >
                {expandFlag ? "HideDetails" : "ShowDetails"}
              </button>
            </div>
            <div className="col-sm">
              <h4>{flightListing[0].departureTime}</h4>
              <span>{flightListing[0].origin}</span>
            </div>
            <div className="col-sm">
              <h4>{flightListing[flightListing.length - 1].arrivalTime}</h4>
              <span>{flightListing[flightListing.length - 1].destination}</span>
            </div>
            <div className="col-sm">
              <h4 className={styles.totalDuration}>
                {flightListing[0].totaDurationofJourney}
              </h4>
              <span>Total Duration</span>
            </div>
            <div className="col-sm">
              <h4 className={styles.redColor}>
                {new Intl.NumberFormat("en-IN", {
                  style: "currency",
                  currency: "INR"
                }).format(price)}
              </h4>
            </div>
            <div className="col-sm">
              {addToFavouriteList && (
                <button
                  className={clsx("btn btn-success", styles.cursorPointer)}
                  onClick={() => {
                    setDisabled(true);
                    addToFavouriteList(flightListing);
                  }}
                  disabled={isDisabled}
                >
                  Favourite
                </button>
              )}
            </div>
            {addRemoveItem && (
              <div className="col-sm">
                <button
                  className={clsx("btn btn-primary", styles.cursorPointer)}
                  onClick={() => addRemoveItem(flightListing, true)}
                >
                  ADD
                </button>
              </div>
            )}
            {addRemoveItem && (
              <div className="col-sm">
                <button
                  className={clsx("btn btn-danger", styles.cursorPointer)}
                  onClick={() => addRemoveItem(flightListing, false, id)}
                >
                  REMOVE
                </button>
              </div>
            )}
          </div>
        </div>
        {expandFlag &&
          _.map(flightListing, (item, index) => {
            return (
              <div>
                <Flightlist
                  data={item}
                  key={item.flightNo + index}
                  multiple={true}
                />
                {!(index % 2) && (
                  <div className={styles.separator}>
                    Layover Time: {flightListing[0].layoverTime}
                  </div>
                )}
              </div>
            );
          })}
      </div>
    </div>
  );
};

MultipleFlights.defaultpropTypes = {
  flightListing: [],
  id: ""
};

MultipleFlights.propTypes = {
  flightListing: PropTypes.any,
  addToFavouriteList: PropTypes.func,
  addRemoveItem: PropTypes.func,
  id: PropTypes.string
};

export default MultipleFlights;
