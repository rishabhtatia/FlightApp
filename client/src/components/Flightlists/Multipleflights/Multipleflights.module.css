.multipleFlights {
  margin: 5px;
  border: 1px solid black;
}
.expandFlagButton {
  color: blue;
  text-decoration: underline;
  border: none;
  background: white;
}
.totalDuration {
  color: green;
}
.redColor {
  color: red;
}
.separator {
  display: flex;
  align-items: center;
  text-align: center;
}
.separator::before,
.separator::after {
  content: "";
  flex: 1;
  border-top: 3px dashed #bbb;
}
.separator::before {
  margin-right: 5px;
}
.separator::after {
  margin-left: 5px;
}
.cursorPointer {
  cursor: pointer;
}
