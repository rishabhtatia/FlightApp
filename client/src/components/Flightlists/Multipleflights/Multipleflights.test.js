import React from 'react';
import renderer from 'react-test-renderer';

import Multipleflights from './Multipleflights';

it('renders when there is one item', () => {
  const item = [
    {
      arrivalTime: '7:10',
      date: '2020/11/01',
      departureTime: '6:10',
      destination: 'Mumbai (BOM)',
      flightNo: 'AI-103',
      name: 'Air India',
      origin: 'Pune (PNQ)',
      price: 2537,
    },
    {
      arrivalTime: '11:30',
      date: '2020/11/01',
      departureTime: '9:00',
      destination: 'Delhi (DEL)',
      flightNo: 'AI-110',
      name: 'Air India',
      origin: 'Mumbai (BOM)',
      price: 4137,
    },
  ];
  const tree = renderer
    .create(<Multipleflights flightListing={item} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
