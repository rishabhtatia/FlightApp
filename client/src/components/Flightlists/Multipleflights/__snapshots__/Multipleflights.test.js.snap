// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is one item 1`] = `
<div
  className="multipleFlights"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <h4>
            Multiple
          </h4>
          <button
            className="expandFlagButton"
            onClick={[Function]}
          >
            ShowDetails
          </button>
        </div>
        <div
          className="col-sm"
        >
          <h4>
            6:10
          </h4>
          <span>
            Pune (PNQ)
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4>
            11:30
          </h4>
          <span>
            Delhi (DEL)
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4
            className="totalDuration"
          >
            5h:20m
          </h4>
          <span>
            Total Duration
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4
            className="redColor"
          >
            â‚¹6,674.00
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
